python/dgl/transforms/functional.py

```diff
@@ -3856,4 +3857,133 @@ def double_radius_node_labeling(g, src, dst):
 
     return F.tensor(z, F.int64)
 
+def shortest_dist(g, root=None, directed=True, return_paths=False):
+    r"""Compute shortest distance and paths on the given graph.
+
+    This function computes the shortest distance between all pairs of nodes
+    or between all nodes and a specified root node. Corresponding shortest
+    paths can be returned and represented with edge IDs. Note that only
+    unweighted cases are allowed.
+
+    Parameters
+    ----------
+    g : DGLGraph
+        The input graph. Must be homogeneous.
+    root : int, optional
+        The root node ID if the graph is provided with a root node.
+        Default: None. If a root node ID is provided, the function returns
+        single-source shortest paths. Otherwise, it returns all-pairs
+        shortest paths.
```

Comment:
```suggestion
        Given a root node ID provided, it returns the shortest distance/path between the root node and all the nodes. Otherwise, it returns the results for all node pairs.
```
I will suggest a simplified version as above.